conference/talks.py

```diff
@@ -23,10 +31,53 @@ def talk(request, talk_slug):
             "social_image_url": request.build_absolute_uri(
                 reverse("conference-talk-social-card-png", kwargs={"slug": talk.slug})
             ),
+            "can_update_talk": can_update_talk,
         },
     )
 
 
+@login_required
+def update_talk(request, talk_slug):
+    """
+    Update talk if you're the author.
+    """
+    talk = get_object_or_404(Talk, slug=talk_slug)
+
+    if not can_user_update_talk(user=request.user, talk=talk):
+        return HttpResponseForbidden()
+
+    talk_update_form = TalkUpdateForm(instance=talk)
+
+    if request.method == "POST":
+        talk_update_form = TalkUpdateForm(request.POST, instance=talk)
+
+        if talk_update_form.is_valid():
+            talk_update_form.save(request.user)
+            messages.success(request, "Talk details updated")
+            return redirect(talk.get_absolute_url())
+        else:
+            messages.error(request, "Please correct the errors below")
+
+    return TemplateResponse(
+        request,
+        "ep19/bs/talks/update_talk.html",
+        {"talk": talk, "talk_update_form": talk_update_form},
+    )
+
+
+def can_user_update_talk(user, talk):
+    """
+    Check if the user can update the given talk.
+    """
+    conf = Conference.objects.current()
+
+    return (
+        talk.created_by == user
+        and talk.status == TALK_STATUS.accepted
+        and not conf.has_finished
```

Comment:
that conf.has_finished part is tricky, depends on how we'd want to handle the slide upload if it happens after the conference, but let's keep it as is for now.

Reply:
Sounds good; the slide upload sounds like a candidate for a separate form + view.